src/blas/backends/rocblas/rocblas_batch.cpp

```diff
@@ -0,0 +1,1382 @@
+inline void gemm_batch(Func func, cl::sycl::queue &queue, transpose transa, transpose transb,
+                       int64_t m, int64_t n, int64_t k, T alpha, cl::sycl::buffer<T, 1> &a,
+                       int64_t lda, int64_t stride_a, cl::sycl::buffer<T, 1> &b, int64_t ldb,
+                       int64_t stride_b, T beta, cl::sycl::buffer<T, 1> &c, int64_t ldc,
+                       int64_t stride_c, int64_t batch_size) {
+    using rocDataType = typename RocEquivalentType<T>::Type;
+    overflow_check(m, n, k, lda, ldb, ldc, stride_a, stride_b, stride_c, batch_size);
+    queue.submit([&](cl::sycl::handler &cgh) {
+        auto a_acc = a.template get_access<cl::sycl::access::mode::read>(cgh);
+        auto b_acc = b.template get_access<cl::sycl::access::mode::read>(cgh);
+        auto c_acc = c.template get_access<cl::sycl::access::mode::read_write>(cgh);
+        onemkl_rocblas_host_task(cgh, queue, [=](RocblasScopedContextHandler &sc) {
+            auto handle = sc.get_handle(queue);
+
+            auto a_ = sc.get_mem<rocDataType *>(a_acc);
+            auto b_ = sc.get_mem<rocDataType *>(b_acc);
+            auto c_ = sc.get_mem<rocDataType *>(c_acc);
+            rocblas_status err;
+            ROCBLAS_ERROR_FUNC(func, err, handle, get_rocblas_operation(transa),
+                               get_rocblas_operation(transb), m, n, k, (rocDataType *)&alpha, a_,
+                               lda, stride_a, b_, ldb, stride_b, (rocDataType *)&beta, c_, ldc,
+                               stride_c, batch_size);
+        });
+    });
+}
+
+#define GEMM_STRIDED_BATCH_LAUNCHER(TYPE, ROCBLAS_ROUTINE)                                         \
+    void gemm_batch(cl::sycl::queue &queue, transpose transa, transpose transb, int64_t m,         \
+                    int64_t n, int64_t k, TYPE alpha, cl::sycl::buffer<TYPE, 1> &a, int64_t lda,   \
+                    int64_t stride_a, cl::sycl::buffer<TYPE, 1> &b, int64_t ldb, int64_t stride_b, \
+                    TYPE beta, cl::sycl::buffer<TYPE, 1> &c, int64_t ldc, int64_t stride_c,        \
+                    int64_t batch_size) {                                                          \
+        gemm_batch(ROCBLAS_ROUTINE, queue, transa, transb, m, n, k, alpha, a, lda, stride_a, b,    \
+                   ldb, stride_b, beta, c, ldc, stride_c, batch_size);                             \
+    }
+
+GEMM_STRIDED_BATCH_LAUNCHER(sycl::half, rocblas_hgemm_strided_batched)
+GEMM_STRIDED_BATCH_LAUNCHER(float, rocblas_sgemm_strided_batched)
+GEMM_STRIDED_BATCH_LAUNCHER(double, rocblas_dgemm_strided_batched)
+GEMM_STRIDED_BATCH_LAUNCHER(std::complex<float>, rocblas_cgemm_strided_batched)
+GEMM_STRIDED_BATCH_LAUNCHER(std::complex<double>, rocblas_zgemm_strided_batched)
+
+#undef GEMM_STRIDED_BATCH_LAUNCHER
+
+void trsm_batch(cl::sycl::queue &queue, side left_right, uplo upper_lower, transpose trans,
+                diag unit_diag, std::int64_t m, std::int64_t n, float alpha,
```

Comment:
Let's be consistent in int64_t usage. (Here and other locations)
```suggestion
                diag unit_diag, int64_t m, int64_t n, float alpha,
```